torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -723,10 +690,34 @@ KinetoEvent::KinetoEvent(
     std::shared_ptr<const torch::profiler::impl::Result> result)
     : result_{result} {
   TORCH_INTERNAL_ASSERT(result != nullptr);
+
+  // Populate Python stack
+  auto parent = result_->parent_.lock();
+  while (parent != nullptr) {
+    parent->visit([&](const auto& i) {
+      if (is_py_fields<decltype(i)>()) {
+        python_stack_.push_back(parent->name());
+      }
+    });
+    parent = parent->parent_.lock();
+  }
 }
 
 bool KinetoEvent::isPythonFunction() const {
-  return result_->kinetoType() == libkineto::ActivityType::PYTHON_FUNCTION;
+  return result_->visit(
+      [](const auto& i) { return is_py_fields<decltype(i)>(); });
+}
+
+const c10::ArrayRef<std::string> KinetoEvent::stack() const {
+  auto get = [&](const auto& i) {
```

Comment:
C++: Just use lambdas, they simplify everything.
Also C++: "This line is missing ` -> auto&` so we made a copy and now ASAN is mad."